src/test/java/com/google/sps/EventTest.java

```diff
@@ -0,0 +1,276 @@
+      .setCategory(CATEGORY1)
+      .setTags(TAGS1)
+      .setDuration(MINUTES_20)
+      .setLocation(LOCATION1)
+      .setLinks(LINKS1)
+      .setFields(FIELDS11)
+      .setInvitedIDs(Arrays.asList(USER_ID2))
+      .build();
+
+    Assert.assertFalse(event1.equals(event2));
+  }
+
+  @Test
+  public void nonEqualityOfEventsWithDifferentParameters() {
+    Event event1 = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .setGCalendarID(G_ID1)
+      .setTitle(TITLE1)
+      .build();
+    
+    Event event2 = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .setGCalendarID(G_ID1)
+      .setTitle(TITLE1)
+      .setDescription(DESCRIPTION1)
+      .setCategory(CATEGORY1)
+      .setTags(TAGS1)
+      .setDuration(MINUTES_20) // different duration
+      .setLocation(LOCATION1)
+      .setLinks(LINKS1)
+      .setFields(FIELDS11)
+      .setInvitedIDs(Arrays.asList(USER_ID2))
+      .build();
+
+    Assert.assertFalse(event1.equals(event2));
+  }
+
+  @Test
+  public void nonEqualityOfEventsDuration() {
+    Event event1 = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .setGCalendarID(G_ID1)
+      .setTitle(TITLE1)
+      .setDescription(DESCRIPTION1)
+      .setCategory(CATEGORY1)
+      .setTags(TAGS1)
+      .setDuration(MINUTES_10) // different duration
+      .setLocation(LOCATION1)
+      .setLinks(LINKS1)
+      .setFields(FIELDS10)
+      .setInvitedIDs(Arrays.asList(USER_ID2))
+      .build();
+    
+    Event event2 = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .setGCalendarID(G_ID1)
+      .setTitle(TITLE1)
+      .setDescription(DESCRIPTION1)
+      .setCategory(CATEGORY1)
+      .setTags(TAGS1)
+      .setDuration(MINUTES_20) // different duration
+      .setLocation(LOCATION1)
+      .setLinks(LINKS1)
+      .setFields(FIELDS11)
+      .setInvitedIDs(Arrays.asList(USER_ID2))
+      .build();
+
+    Assert.assertFalse(event1.equals(event2));
+  }
+
+  @Test
+  public void joinInvitedEvent() {
+    Event actual_event = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .setInvitedIDs(Arrays.asList(USER_ID2))
+      .build();
+
+    actual_event.joinEvent(USER_ID2);
+
+    Event expected_event = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .setJoinedIDs(new ArrayList<String>(Arrays.asList(USER_ID2)))
+      .build();
+
+    Assert.assertEquals(expected_event.equals(actual_event), true);
+  }
+
+  @Test
+  public void joinNotInvitedEvent() {
+    Event actual_event = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .build();
+
+    actual_event.joinEvent(USER_ID2);
+
+    Event expected_event = Event.newBuilder()
+      .setID(EVENT_ID1)
+      .setOwnerID(USER_ID1)
+      .build();
+
+    Assert.assertEquals(expected_event.equals(actual_event), true);
+  }
+
+  @Test
+  public void joinOwnEvent() {
```

Comment:
This is kind of misleading because it is no different at the moment whether you own the event or you are not invited to the event. Maybe return a bool in joinEvent to see the difference?